### ISSUE TITLE
Are CTS relationship endpoints being looked up in correct location?

### ISSUE THREAD

**Issue by cmgrote** [odpi/egeria#2074]
It appears the `TestSupportedRelationshipLifecycle` looks up the endpoint EntityDef's from the types found in the repository itself: should it be looking up against these, or against the known open metadata types?

```java
EntityDef end1Type = entityDefs.get(relationshipDef.getEndDef1().getEntityType().getName());
EntityDetail end1 = this.addEntityToRepository(workPad.getLocalServerUserId(), metadataCollection, end1Type);
EntityDef end2Type = entityDefs.get(relationshipDef.getEndDef2().getEntityType().getName());
EntityDetail end2 = this.addEntityToRepository(workPad.getLocalServerUserId(), metadataCollection, end2Type);
```

I ask as in a repository that only implements the "leaf" nodes of the inheritance hierarchy (like IGC which implements things like `Database`, `Person`, etc but not their supertypes), this results in NPE's for relationships whose endpoint definitions refer to more abstract types (like `Asset`, `DataSet` or `ActorProfile`) -- that is, those EntityDefs are not implemented in the repository, just their subtypes (including all the properties of the supertype).

Hence I wonder if these lookups should be against the open metadata types archive rather than the set of definitions available in the repository itself?

(Which I believe would also ensure that if the repository has "overridden" these definitions for any reason, that we are still testing they adhere to the open metadata types themselves rather than a repository's own type implementation?)

**Comment by grahamwallis** [comment]
I think there may be two aspects to this. 

One is that the test case should only ask the repository connector to instantiate ends using entity types that the repository supports. The repository can legitimately verify that it supports a relationship type that has end definitions specifying types that are more abstract than any entity types supported by the repository - and when the CTS needs to instantiate a relationship of that type it (the CTS) should hunt for a sub-type that *is* supported by the repository.

The other is that, as you say, the CTS should be getting the type definition from the active set, not from what was returned by the repository connector.

**Comment by cmgrote** [comment]
Resolved by #2093
